cmd/lotus-storage-miner/proving.go

```diff
@@ -132,67 +121,61 @@ var provingInfoCmd = &cli.Command{
 			return xerrors.Errorf("getting chain head: %w", err)
 		}
 
-		cd, err := api.StateMinerProvingDeadline(ctx, maddr, head.Key())
+		mact, err := api.StateGetActor(ctx, maddr, head.Key())
 		if err != nil {
-			return xerrors.Errorf("getting miner info: %w", err)
+			return err
 		}
 
-		deadlines, err := api.StateMinerDeadlines(ctx, maddr, head.Key())
+		stor := store.ActorStore(ctx, apibstore.NewAPIBlockstore(api))
+
+		mas, err := miner.Load(stor, mact)
 		if err != nil {
-			return xerrors.Errorf("getting miner deadlines: %w", err)
+			return err
 		}
 
-		fmt.Printf("Miner: %s\n", color.BlueString("%s", maddr))
-
-		var mas miner.State
-		{
-			mact, err := api.StateGetActor(ctx, maddr, types.EmptyTSK)
-			if err != nil {
-				return err
-			}
-			rmas, err := api.ChainReadObj(ctx, mact.Head)
-			if err != nil {
-				return err
-			}
-			if err := mas.UnmarshalCBOR(bytes.NewReader(rmas)); err != nil {
-				return err
-			}
+		cd, err := api.StateMinerProvingDeadline(ctx, maddr, head.Key())
+		if err != nil {
+			return xerrors.Errorf("getting miner info: %w", err)
 		}
 
-		parts := map[uint64][]*miner.Partition{}
-		for dlIdx := range deadlines {
-			part, err := api.StateMinerPartitions(ctx, maddr, uint64(dlIdx), types.EmptyTSK)
-			if err != nil {
-				return xerrors.Errorf("getting miner partition: %w", err)
-			}
-
-			parts[uint64(dlIdx)] = part
-		}
+		fmt.Printf("Miner: %s\n", color.BlueString("%s", maddr))
 
 		proving := uint64(0)
 		faults := uint64(0)
 		recovering := uint64(0)
+		curDeadlineSectors := uint64(0)
 
-		for _, partitions := range parts {
-			for _, partition := range partitions {
-				sc, err := partition.Sectors.Count()
-				if err != nil {
-					return xerrors.Errorf("count partition sectors: %w", err)
+		if err := mas.ForEachDeadline(func(dlIdx uint64, dl miner.Deadline) error {
```

Comment:
This is a case where we should _really_ consider adding a dedicated API method.

Reply:
We could add those counts to `StateMinerSectorCount(): MinerSectors`

Reply:
You know, I'm not actually sure now. We need to compute the a proof for the _future_ so we don't want to attach that to the current state.